scripts/globals/chocobo_riding_game.lua

```diff
@@ -0,0 +1,209 @@
+        end
+    end
+
+    return false
+end
+
+-- Get the event param to tell the player which city race they are offering
+xi.chocoboGame.renterOnTrigger = function(player, destination, eventSucceed)
+    local zoneId     = player:getZone():getID()
+    local eventParam = raceData[zoneId][destination].eventParam
+
+    -- Temp destination var until player confirms race
+    player:setLocalVar('[ChocoGame]DestCity', destination)
+    player:startEvent(eventSucceed, -3, 0, 0, eventParam)
+end
+
+-- Apply race vars, check for csid and option is done in chocobo.lua
+xi.chocoboGame.renterOnEventFinish = function(player, option)
+    if option == 0 then
+        local zoneId     = player:getZone():getID()
+        local destCity   = player:getLocalVar('[ChocoGame]DestCity')
+
+        player:setCharVar('[ChocoGame]StartingCity', zoneId)
+        player:setCharVar('[ChocoGame]DestCity', destCity)
+        player:setCharVar('[ChocoGame]NextEntryTime', 1, NextConquestTally())
+        player:setCharVar('[ChocoGame]StartTime', os.time())
+    else
+        player:setLocalVar('[ChocoGame]DestCity', 0)
+    end
+end
+
+-- Determine which tier of award the player gets based off their time
+xi.chocoboGame.rewardCheck = function(startingCity, destCity, clearTime)
+    local raceTimes = raceData[startingCity][destCity].raceTimes
+    local reward    = 0
+    local jeuno     =
+    {
+        xi.zone.ROLANBERRY_FIELDS,
+        xi.zone.SAUROMUGUE_CHAMPAIGN,
+        xi.zone.BATALLIA_DOWNS
+    }
+
+    -- Jeuno paths only have two rewards
+    if utils.contains(destCity, jeuno) then
+        if clearTime < raceTimes[1] then -- Gold
+            reward = raceData[startingCity][destCity].reward
+        else -- Bronze
+            reward = defaultRewards[2]
+        end
+    else
+        if clearTime < raceTimes[1] then -- Gold
+            reward = raceData[startingCity][destCity].reward
+        elseif clearTime >= raceTimes[1] and clearTime <= raceTimes[2] then -- Silver
+            reward = defaultRewards[1]
+        else -- Bronze
+            reward = defaultRewards[2]
+        end
+    end
+
+    return reward
+end
+
+-- Player reached finish line, handle event messaging
+xi.chocoboGame.onTriggerAreaEnter = function(player)
+    local destCity = player:getCharVar('[ChocoGame]DestCity')
+
+    if player:getZoneID() == destCity then
+        local clearTime = os.time() - player:getCharVar('[ChocoGame]StartTime')
+        local startingCity = player:getCharVar('[ChocoGame]StartingCity')
+        local recordId = GetServerVariable('[ChocoGame][RecordHolder]'..startingCity..'+'..destCity)
+        local recordTime = GetServerVariable('[ChocoGame][RecordTime]'..startingCity..'+'..destCity)
+
+        -- Parse record name
+        local recordName = ""
+        if recordId ~= 0 then
+            recordName = GetPlayerByID(recordId):getName()
+        end
+
+        -- Race complete event if a record for this path exists
+        if recordName then
```

Comment:
As previously, might want to confirm that an empty string is interpreted as false

Reply:
My testing showed this working correctly, the messaging from the NPC changed on my test depending on if a record existed or not